cmd/lncli/commands.go

```diff
@@ -672,51 +679,110 @@ func closeChannel(ctx *cli.Context) error {
 		req.ChannelPoint.OutputIndex = 0
 	}
 
-	stream, err := client.CloseChannel(ctxb, req)
+	closingTxid, err := closeChannelRequest(ctx, req)
 	if err != nil {
 		return err
 	}
 
+	printJSON(struct {
+		ClosingTxid string `json:"closing_txid"`
+	}{
+		ClosingTxid: closingTxid,
+	})
+
+	return nil
+}
+
+func closeChannelRequest(ctx *cli.Context, req *lnrpc.CloseChannelRequest) (string, error) {
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	stream, err := client.CloseChannel(context.Background(), req)
+	if err != nil {
+		return "", err
+	}
+
+	var closingTxid string
+
 	for {
 		resp, err := stream.Recv()
 		if err == io.EOF {
-			return nil
+			return closingTxid, nil
 		} else if err != nil {
-			return err
+			return "", err
 		}
 
 		switch update := resp.Update.(type) {
 		case *lnrpc.CloseStatusUpdate_ClosePending:
 			closingHash := update.ClosePending.Txid
 			txid, err := chainhash.NewHash(closingHash)
 			if err != nil {
-				return err
+				return "", err
 			}
 
-			printJSON(struct {
-				ClosingTXID string `json:"closing_txid"`
-			}{
-				ClosingTXID: txid.String(),
-			})
+			closingTxid = txid.String()
 
 			if !ctx.Bool("block") {
-				return nil
+				return closingTxid, nil
 			}
-
 		case *lnrpc.CloseStatusUpdate_ChanClose:
 			closingHash := update.ChanClose.ClosingTxid
 			txid, err := chainhash.NewHash(closingHash)
 			if err != nil {
-				return err
+				return "", err
 			}
 
-			printJSON(struct {
-				ClosingTXID string `json:"closing_txid"`
-			}{
-				ClosingTXID: txid.String(),
-			})
+			closingTxid = txid.String()
+		}
+	}
+}
+
+func closeAllChannels(ctx *cli.Context) error {
+	ctxb := context.Background()
+	client, cleanUp := getClient(ctx)
+	defer cleanUp()
+
+	listReq := &lnrpc.ListChannelsRequest{}
+	openChannels, err := client.ListChannels(ctxb, listReq)
+	if err != nil {
+		return fmt.Errorf("unable to fetch open channels: %v", err)
+	}
+
+	txids := make([]string, len(openChannels.Channels))
+	for i, channel := range openChannels.Channels {
+		s := strings.Split(channel.ChannelPoint, ":")
+		txid, err := chainhash.NewHashFromStr(s[0])
+		if err != nil {
+			continue
+		}
+		index, err := strconv.ParseUint(s[1], 10, 32)
+		if err != nil {
+			continue
+		}
+
+		req := &lnrpc.CloseChannelRequest{
+			ChannelPoint: &lnrpc.ChannelPoint{
+				FundingTxid: txid[:],
```

Comment:
Note that this'll need to be modified to use the `oneof` field once the other PR fixing the issue w.r.t force closing over REST if fixed.